### DISCUSSION TITLE
How to manage variants with multiple selectable attributes ? E.g using two select

### DISCUSSION THREAD

**Discussion by DidoMarchet** [saleor/saleor#10566] [General]
Hi,
we need to manage variants with multiple selectable attributes (color, size).
This behaviour will create a matrix of variants.

We want to display two select, once per attribute.

Variants are displayed as couple of attributes.

<img width="129" alt="Screenshot 2022-09-07 at 10 24 29" src="https://user-images.githubusercontent.com/38940340/188830028-bd3fa9ca-605b-40db-b695-c224836ff607.png">

I cannot figure out a easy way to split in two select this matrix and select a variant in base of the value of the select.

Int he old Saleor storefront demo there was some example but in the new no.
I want to avoid long solution such as reduce and filter the variant array if possible.

Do you have any suggestion ?

Really thanks in advance and regards,

Davide

**Comment by peelar** [comment]
Hello @DidoMarchet 👋 

I will need some help to better understand what your problem is. Are you using our Dashboard, or is this a custom implementation? Where does your screenshot come from? Are you struggling with the JavaScript side of things or the API?

**Reply by DidoMarchet** [reply]
Hi @peelar,
sorry for the incomplete request.
This is not an issue but just thought

I have a product with some variants.

<img width="868" alt="Screenshot 2022-09-08 at 11 17 25" src="https://user-images.githubusercontent.com/38940340/189084998-92604254-1b3e-4cf9-b6e4-affe95a2e51b.png">

These variants are made up of two attributes 

<img width="928" alt="Screenshot 2022-09-08 at 11 18 03" src="https://user-images.githubusercontent.com/38940340/189085210-0c017e0b-4bfb-4649-926c-59ae41998842.png">

In my frontend I need to decouple these attributes in two input select.
One select with the attribute Lens Color (Colore Lenti)  and one with attribute  Frame Size (Taglia Montatura).

When I select for example lens color "Red" and frame size "M" I need to write a little bit of code (and it's ok) to get the correct variantId mattching the values of the selected attributes.

My question is if there's a simple way to do that but I checked the code of the old demo storefront where some products have variants with multiple attributes and the solutions is the same of mine.

Maybe could be useful add a query to a product that giving some variant attribute ids returns the matched variantId.

Thanks in advance and kind regards,
Davide
